src/SignalR/clients/ts/signalr/webpack.webworker.config.js

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+const baseConfig = require("../webpack.config.base");
+module.exports = baseConfig(__dirname, "signalr", {
+    // These are only used in Node environments
+    // so we tell webpack not to pull them in for the browser
+    target: "webworker",
+    platformDist: "webworker",
+    externals: [
+        "websocket",
+        "eventsource",
+        "request"
+    ]
+});
```

Comment:
Why do we need this in addition to the main webpack.config.js? It looks like it's just changing the output directory even though the client source code is identical and all the platform checks are done at runtime. What am I missing?

Reply:
Forgive my ignorance here. Do we actually need a new target and output directory? Why can't we have a single target that works in both a normal browser context and a webworker context?

Reply:
Because the global window object is not available in WebWorker and if we use build script from `web` inside of webworker the SignalR script execution failed because window object is not available.
More over there are some differences between build target for `webworker` and `web`
https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsApply.js#L98